tsuru/client/metadata.go

```diff
@@ -14,93 +13,135 @@ import (
 	"github.com/tsuru/go-tsuruclient/pkg/tsuru"
 	"github.com/tsuru/tsuru-client/tsuru/formatter"
 	"github.com/tsuru/tsuru/cmd"
-	appTypes "github.com/tsuru/tsuru/types/app"
 )
 
 const metadataSetValidationMessage = `You must specify metadata in the form "NAME=value" with the specified type.
 
 Example:
 
-  tsuru app-metadata-set -a APPNAME -t label NAME=value OTHER_NAME="value with spaces" ANOTHER_NAME='using single quotes'
-  tsuru app-metadata-set -a APPNAME -t annotation NAME=value OTHER_NAME="value with spaces" ANOTHER_NAME='using single quotes'
+  tsuru metadata-set <-a APPNAME | -j JOBNAME> -t label NAME=value OTHER_NAME="value with spaces" ANOTHER_NAME='using single quotes'
+  tsuru metadata-set <-a APPNAME | -j JOBNAME> -t annotation NAME=value OTHER_NAME="value with spaces" ANOTHER_NAME='using single quotes'
 
 `
 
 var allowedTypes = []string{"label", "annotation"}
 
+type JobOrApp struct {
+	Type string
+	val  string
+	fs   *gnuflag.FlagSet
+}
+
+func (c *JobOrApp) validate() error {
+	appName := c.fs.Lookup("app").Value.String()
+	jobName := c.fs.Lookup("job").Value.String()
+	if appName == "" && jobName == "" {
+		return errors.New("job name or app name is required")
+	}
+	if appName != "" && jobName != "" {
+		return errors.New("please use only one of the -a/--app and -j/--job flags")
+	}
+	if appName != "" {
+		c.Type = "app"
+		c.val = appName
+		return nil
+	}
+	c.Type = "job"
+	c.val = jobName
+	return nil
+}
+
+func (c *JobOrApp) getMetdata(apiClient *tsuru.APIClient) (tsuru.Metadata, error) {
+	if c.Type == "job" {
+		job, _, err := apiClient.JobApi.GetJob(context.Background(), c.val)
+		if err != nil {
+			return tsuru.Metadata{}, err
+		}
+		return job.Job.Metadata, nil
+	}
+	app, _, err := apiClient.AppApi.AppGet(context.Background(), c.val)
+	if err != nil {
+		return tsuru.Metadata{}, err
+	}
+	return app.Metadata, nil
+}
+
+func (c *JobOrApp) setMetdata(apiClient *tsuru.APIClient, metadata tsuru.Metadata, noRestart bool) (*http.Response, error) {
```

Comment:
little typo, change to setMetadata @morpheu

Reply:
thanks!